openex-api/src/main/java/io/openex/config/AppSecurityConfig.java

```diff
@@ -26,165 +29,398 @@
+        return getAttributes();
+      }
+
+      @Override
+      public OidcUserInfo getUserInfo() {
+        return OidcUserInfo.builder().name(getName()).email(getEmail()).build();
+      }
+
+      @Override
+      public OidcIdToken getIdToken() {
+        return null;
+      }
+
+      @Override
+      public Map<String, Object> getAttributes() {
+        HashMap<String, Object> attributes = new HashMap<>();
+        attributes.put("id", loginUser.getId());
+        attributes.put("name", loginUser.getFirstname() + " " + loginUser.getLastname());
+        attributes.put("email", loginUser.getEmail());
+        return attributes;
+      }
+
+      @Override
+      public Collection<? extends GrantedAuthority> getAuthorities() {
+        List<SimpleGrantedAuthority> roles = new ArrayList<>();
+        roles.add(new SimpleGrantedAuthority(ROLE_USER));
+        if (loginUser.isAdmin()) {
+          roles.add(new SimpleGrantedAuthority(ROLE_ADMIN));
+        }
+        return roles;
+      }
+
+      @Override
+      public String getName() {
+        return loginUser.getFirstname() + " " + loginUser.getLastname();
+      }
+
+    };
+  }
+
+  public OAuth2User oAuth2UserManagement(OAuth2AccessToken accessToken, ClientRegistration clientRegistration,
+      OAuth2User user) {
+    User loginUser = userOauth2Management(accessToken, clientRegistration, user);
+    return new OpenExOAuth2User() {
+      @Override
+      public Map<String, Object> getAttributes() {
+        HashMap<String, Object> attributes = new HashMap<>();
+        attributes.put("id", loginUser.getId());
+        attributes.put("name", loginUser.getFirstname() + " " + loginUser.getLastname());
+        attributes.put("email", loginUser.getEmail());
+        return attributes;
+      }
+
+      @Override
+      public String getId() {
+        return loginUser.getId();
+      }
+
+      @Override
+      public Collection<? extends GrantedAuthority> getAuthorities() {
+        List<SimpleGrantedAuthority> roles = new ArrayList<>();
+        roles.add(new SimpleGrantedAuthority(ROLE_USER));
+        if (loginUser.isAdmin()) {
+          roles.add(new SimpleGrantedAuthority(ROLE_ADMIN));
+        }
+        return roles;
+      }
+
+      @Override
+      public boolean isAdmin() {
+        return false;
+      }
+
+      @Override
+      public String getName() {
+        return loginUser.getFirstname() + " " + loginUser.getLastname();
+      }
+
+    };
+  }
+
+  public Saml2Authentication saml2UserManagement(
+      Saml2Authentication authentication) {
+    Saml2AuthenticatedPrincipal user = (Saml2AuthenticatedPrincipal) authentication.getPrincipal();
+    User loginUser = userSaml2Management(user);
+
+    List<SimpleGrantedAuthority> roles = new ArrayList<>();
+    roles.add(new SimpleGrantedAuthority(ROLE_USER));
+    if (loginUser.isAdmin()) {
+      roles.add(new SimpleGrantedAuthority(ROLE_ADMIN));
+    }
+
+    return new Saml2Authentication(new OpenExSaml2User() {
```

Comment:
OpenExSaml2User can be a class to move all this next line and be more readable
What do you think ?